### PR TITLE
Allow merging vectors into all builder types

### LINKED ISSUE #173: Support all Builders for Vectorization
Builders which do not have explicit vectorization should unroll the vector because merging values.

### PR DESCRIPTION
**Pull request by mateiz**
Also has some minor improvements for debug messages. Fixes #173.

### CODE REVIEW

File: weld/bin/repl.rs [hunk 1]
```diff
@@ -128,7 +128,7 @@ fn main() {
         }
 
         if let Err(ref e) = program {
-            println!("Error during parsing: {:?}", e);
+            println!("Error during parsing: {}", e);
```
Comment (mateiz) [inline weld/bin/repl.rs]: This is just to prevent some weird escaped quotes from appearing here

File: weld/llvm.rs [hunk 2]
```diff
@@ -1126,6 +1127,7 @@ impl LlvmGenerator {
 
     /// Generate code for a single statement, appending it to the code in a FunctionContext.
     fn gen_statement(&mut self, statement: &Statement, func: &SirFunction, ctx: &mut FunctionContext) -> WeldResult<()> {
+        ctx.code.add(format!("; {}", statement));
```
Comment (mateiz) [inline weld/llvm.rs]: This is amazingly useful -- look at the generated code after!

File: weld/vectorizer.rs [hunk 3]
```diff
@@ -180,30 +199,27 @@ fn vectorizable(for_loop: &Expr<Type>) -> WeldResult<HashSet<Symbol>> {
...
-                                let mut getfield_passed = false;
-                                let ref elem_param = params[2];
-                                if let Ident(ref name) = expr.kind {
-                                    if elem_param.name == *name {
-                                       getfield_passed = true; 
-                                    }
-                                }
-                                passed = getfield_passed;
-                            }
-                            Merge{ .. } => {},
```
Comment (mateiz) [inline weld/vectorizer.rs]: @sppalkia is it bad to remove this? Getting rid of it allows more code to be vectorized, e.g. stuff that is creating intermediate structs for use in dictmerger, etc.

File: weld/llvm.rs (hunk 2 above)
Comment (sppalkia) [inline weld/llvm.rs]: What does this do?

Comment (sppalkia) [inline weld/llvm.rs]: Ah never mind - it just appends the SIR expression

File: weld/vectorizer.rs (hunk 3 above)
Comment (sppalkia) [inline weld/vectorizer.rs]: This should be okay...it works with nested structures and everything right? So basically, the behavior is to broadcast each literal in the struct?

**Comment by mateiz** [comment]
Sure, in fact it did vectorize a lot of the existing tests because they failed without some fixes, but let me check for that explicitly.

File: weld/vectorizer.rs [hunk 4]
```diff
@@ -14,12 +14,42 @@ use std::collections::HashSet;
...
+pub fn is_simd(ty: &Type) -> bool {
+    match *ty {
+        Simd(_) => true,
+        Struct(ref fields) => fields.iter().all(is_simd),
+        _ => false
+    }
+}
+
+/// Get the scalar type corresponding to a SIMD type.
+pub fn scalar_type(simd_type: &Type) -> WeldResult<Type> {
```
Comment (sppalkia) [inline weld/vectorizer.rs]: Btw, why is this here instead of being a method on `Type`? Same with `is_simd`

Comment (mateiz) [inline weld/vectorizer.rs]: Because `vectorized_type` was here. It may be OK to move them, but then we should move all 3 and we would probably want to make `vectorized_type` to return a WeldResult too to be consistent.

**Comment by mateiz** [comment]
FYI this is failing because of the issue in #180 because we now vectorize the appender-based code in that test.
